scripts/get_status.py

```diff
@@ -1,124 +1,77 @@
-                return run_status['details']['action_statuses'][0]['state_name']
-            elif det.get('code') == 'FlowStarting':
-                return None
-    except Exception:
-        print(traceback.format_exc(), file=sys.stderr)
+def get_run_url(run_id: str):
+    return f"https://app.globus.org/runs/{run_id}"
 
-    print(f'BUG ENCOUNTERED! An unexpected status was returned by the flows '
-          f'service, a dump of the response is listed below. Please send this '
-          f'to us so we can fix it. Thanks!', file=sys.stderr)
-    pprint.pprint(run_status, stream=sys.stderr)
-    sys.exit(2)
 
+def get_run_logs(flows_client: globus_sdk.FlowsClient, run_id: str):
+    return flows_client.paginated.get_run_logs(args.run_id, limit=30).items()
 
-def state_has_completed(flow_state, flow_states, run_status):
-    current_state = get_current_state_name(run_status)
-    if current_state is None:
-        return False
-    return flow_states.index(current_state) > flow_states.index(flow_state)
 
+def is_target_state_in_run_logs(flows_client: globus_sdk.FlowsClient, run_id: str, target_step: str):
+    for pagelog in get_run_logs(flows_client, run_id):
+        if pagelog.get("code") in ["PassCompleted", "ActionCompleted"]:
+            step = pagelog["details"]["state_name"]
+            if step == target_step:
+                return True
+    return False
 
-def check_run_status(run_status):
-    url = f'https://app.globus.org/runs/{run_status["run_id"]}'
-    if run_status['status'] == 'FAILED':
-        raise RunFailed(f'Run Failed: {url}')
-    elif run_status['status'] == 'SUCCEEDED':
-        raise RunSucceeded(f'Run Succeeded: {url}')
 
-
-if __name__ == '__main__':
+if __name__ == "__main__":
     args = arg_parse()
-
-    callback_login_manager = None
-    if CLIENT_ID and CLIENT_SECRET:
-        callback_login_manager = CallbackLoginManager({}, callback=callback)
-
-    if args.gpu:
-        main_flow = XPCSBoost(login_manager=callback_login_manager)
-    else:
-        main_flow = XPCSEigen(login_manager=callback_login_manager)
-
-    # Fetch state names in a loose ordering, depth first
-    flow_states = [state_name for state_name, _ in iter_flow(main_flow.flow_definition)]
-
-    if args.step not in flow_states:
-        print(f'"{args.step}" is not valid, please choose from: {", ".join(flow_states)}')
-        sys.exit(-1)
-
-    try:
-        while True:
-            status = main_flow.get_status(args.run_id)
-            check_run_status(status)
-            if state_has_completed(args.step, flow_states, status):
-                print(f'Step {args.step}: Completed')
-                sys.exit(0)
-            time.sleep(args.interval)
-    except RunSucceeded as rs:
-        print(str(rs))
-        sys.exit(0)
-    except (RunFailed) as e:
-        print(f'{e.__class__.__name__}: {str(e)}', file=sys.stderr)
-        sys.exit(1)
\ No newline at end of file
+    fc = get_flows_client()
+
+    while True:
```

Comment:
Good plan, I added a timeout that releases by default after an hour. This could cause unexpected problems if DM prefers to start new flows while runs remain active, so that's something to be aware of too.